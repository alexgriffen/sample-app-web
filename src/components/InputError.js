import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import Input from '@material-ui/core/Input';
import { withStyles } from '@material-ui/core/styles';
import InputAdornment from '@material-ui/core/InputAdornment';
import Icon from '@material-ui/core/Icon';
import { colors } from '../utils/Colors';

export const INPUT_TYPES = {
  TEXT: 'text',
  PASSWORD: 'password',
};
const styles = () => ({
  root: {
    borderBottom: `2px solid ${ colors.lightGray }`,
  },
  rootError: {
    borderBottom: `2px solid ${ colors.slRed }`,
  },
  inputError: {
    color: colors.slRed,
  },
  icon: {
    color: colors.slRed,
  },
});

const { object, string, bool, func } = PropTypes;

class InputError extends Component {
  static propTypes = {
    classes: object.isRequired,
    dataTest: string,
    error: bool,
    onChange: func.isRequired,
    placeholder: string,
    testID: string,
    type: string,
  }

  static defaultProps = {
    error: false,
    placeholder: 'Placeholder',
    testID: 'test-id',
    type: INPUT_TYPES.TEXT,
  };

  render() {
    const { classes, dataTest, error, testID, onChange, placeholder, type } = this.props;
    const endAdornment = (error ? (
      <InputAdornment>
        <Icon style={ { color: colors.slRed } }>
          cancel
        </Icon>
      </InputAdornment>
    ) : null);

    return (
      <Input
        classes={ {
          root: error ? classes.rootError : classes.root,
          // `' form-input'` is there to make it backwards compatible,
          // it doesn't not have an other function
          input: (error ? classes.inputError : {}) + ' form-input',
        } }
        disableUnderline
        inputProps={ {
          placeholder: placeholder,
          ...(dataTest ? { 'data-test': dataTest } : {}),
        } }
        fullWidth
        id={ testID }
        onChange={ onChange }
        endAdornment={ endAdornment }
        type={ type }
      />
    );
  }
}

export default withStyles(styles)(InputError);
