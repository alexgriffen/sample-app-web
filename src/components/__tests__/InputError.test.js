import React from 'react';
import { shallow } from 'enzyme/build';
import InputError, { INPUT_TYPES } from '../InputError';

let props;

describe('InputError', () => {
  beforeEach(() => {
    props = {
      onChange: jest.fn(),
    };
  });

  it('should render correctly with the required options', () => {
    const component = shallow(<InputError { ...props } />).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render with full options', () => {
    const component = shallow(
      <InputError
        { ...props }
        dataTest="data-test-string"
        placeholder="placeholder-string"
        testID="testid-string"
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render an error component', () => {
    const component = shallow(
      <InputError
        { ...props }
        error
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render secure input', () => {
    const component = shallow(
      <InputError
        { ...props }
        type={INPUT_TYPES.PASSWORD}
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });
});
