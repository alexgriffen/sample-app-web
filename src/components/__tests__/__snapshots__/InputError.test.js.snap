// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError should render an error component 1`] = `
<WithStyles(Input)
  classes={
    Object {
      "input": "InputError-inputError-3 form-input",
      "root": "InputError-rootError-2",
    }
  }
  disableUnderline={true}
  endAdornment={
    <WithStyles(WithFormControlContext(InputAdornment))>
      <WithStyles(Icon)
        style={
          Object {
            "color": "#E2231A",
          }
        }
      >
        cancel
      </WithStyles(Icon)>
    </WithStyles(WithFormControlContext(InputAdornment))>
  }
  fullWidth={true}
  id="test-id"
  inputProps={
    Object {
      "placeholder": "Placeholder",
    }
  }
  onChange={[MockFunction]}
  type="text"
/>
`;

exports[`InputError should render correctly with the required options 1`] = `
<WithStyles(Input)
  classes={
    Object {
      "input": "[object Object] form-input",
      "root": "InputError-root-1",
    }
  }
  disableUnderline={true}
  endAdornment={null}
  fullWidth={true}
  id="test-id"
  inputProps={
    Object {
      "placeholder": "Placeholder",
    }
  }
  onChange={[MockFunction]}
  type="text"
/>
`;

exports[`InputError should render secure input 1`] = `
<WithStyles(Input)
  classes={
    Object {
      "input": "[object Object] form-input",
      "root": "InputError-root-1",
    }
  }
  disableUnderline={true}
  endAdornment={null}
  fullWidth={true}
  id="test-id"
  inputProps={
    Object {
      "placeholder": "Placeholder",
    }
  }
  onChange={[MockFunction]}
  type="password"
/>
`;

exports[`InputError should render with full options 1`] = `
<WithStyles(Input)
  classes={
    Object {
      "input": "[object Object] form-input",
      "root": "InputError-root-1",
    }
  }
  disableUnderline={true}
  endAdornment={null}
  fullWidth={true}
  id="testid-string"
  inputProps={
    Object {
      "data-test": "data-test-string",
      "placeholder": "placeholder-string",
    }
  }
  onChange={[MockFunction]}
  type="text"
/>
`;
